src/class/hid/hid_device.c

```diff
@@ -202,10 +203,33 @@ bool hidd_control_request(uint8_t rhport, tusb_control_request_t const * p_reque
     uint8_t const desc_index = tu_u16_low (p_request->wValue);
     (void) desc_index;
 
-    if (p_request->bRequest == TUSB_REQ_GET_DESCRIPTOR && desc_type == HID_DESC_TYPE_REPORT)
+    if (p_request->bRequest == TUSB_REQ_GET_DESCRIPTOR && desc_type == HID_DESC_TYPE_HID)
+    {
+      // FIXME: Should check which is the active configuration, but no callback in usbd currently exists to do that
+      TU_VERIFY(p_hid->hid_descriptor != NULL);
+      tusb_desc_configuration_t const* desc_cfg =

```

Comment:
@pigrew we can just return the p_hid->hid_descriptor now, a simple 
`tud_control_xfer(rhport, p_request, (void*) p_hid->hid_descriptor, p_hid->hid_descriptor->bLength)`

will work

Reply:
I remember typing that line... I must have gotten distracted.